### PR TITLE
SLS-591 Only check history store versions in the version cursor if it is open for the btree

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/cursor/cur_version.c [hunk 1]
```diff
@@ -721,12 +722,12 @@ __wt_curversion_open(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *owner
     WT_ERR(version_cursor->file_cursor->close(version_cursor->file_cursor));
     WT_ERR(__wt_open_cursor(session, uri, cursor, file_cursor_cfg, &version_cursor->file_cursor));
 
-    /* Open the history store cursor for operations on the regular history store .*/
-    if (F_ISSET(S2C(session), WT_CONN_HS_OPEN)) {
+    /* Open the history store cursor for btrees may have data in the history store.*/
```
Comment (korteland) [inline src/cursor/cur_version.c:725]: Should this be `... for btrees that may have ...`?

**Review by korteland — APPROVED** [review]
LGTM modulo comment

**Comment by korteland** [comment]
One high level question - the ticket originally wanted an option for the version cursor, I assume this new behaviour (of not enabling the history store for in-memory trees) replaces that?

**Comment by quchenhao** [comment]
> One high level question - the ticket originally wanted an option for the version cursor, I assume this new behaviour (of not enabling the history store for in-memory trees) replaces that?

Yes, that's correct. We should be able to change behaviours based on the types of btree we are dealing with.

File: src/cursor/cur_version.c (hunk 1 above)
Comment (quchenhao) [reply src/cursor/cur_version.c]: fixed

**Pull request merged**